### PR TITLE
[DUOS-2823] Add delete modal dialog for accepted DAC datasets

### PR DESCRIPTION
**Pull request by fboulnois**
### Addresses

https://broadworkbench.atlassian.net/browse/DUOS-2823

### Summary

Add delete modal dialog for accepted DAC datasets

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/components/dac_dataset_table/DACDatasetApprovalStatus.js [hunk 1]
```diff
@@ -1,14 +1,33 @@
 import React, {useState} from 'react';
-import {DAC} from '../../libs/ajax';
+import {DAC, DataSet} from '../../libs/ajax';
 import {Link} from 'react-router-dom';
 import {isNil} from 'lodash/fp';
 import Button from '@mui/material/Button';
 import ReactTooltip from 'react-tooltip';
 import style from '../../pages/DACDatasets.module.css';
+import { ConfirmationDialog } from '../modals/ConfirmationDialog';
+import IconButton from '@mui/material/IconButton';
```
Comment (aarohinadkarni) [inline src/components/dac_dataset_table/DACDatasetApprovalStatus.js]: Can we delete the imports on lines 9 & 10 if they aren't being used?

File: src/components/dac_dataset_table/DACDatasetApprovalStatus.js [hunk 2]
```diff
@@ -1,14 +1,33 @@
...
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const handleAction = (datasetId) => {
+    setOpen(false);
+    DataSet.deleteDataset(datasetId).then(() => {
+      window.location.reload();
+    });
+  }
```
Comment (rushtong) [inline src/components/dac_dataset_table/DACDatasetApprovalStatus.js]: This function should only be called if the dataset is deletable. In the case that it isn't, the user will get an error:
```
{"message":"Dataset is in use and cannot be deleted.","code":400}
```
Look for the  `"deletable": false` property on the dataset to check this condition.

**Review by rushtong — CHANGES_REQUESTED** [review]
Aside from the conditional check needed, the code looks good!

File: src/components/dac_dataset_table/DACDatasetApprovalStatus.js (hunk 2 above)
Comment (fboulnois) [reply src/components/dac_dataset_table/DACDatasetApprovalStatus.js]: I was not aware of this property! Good catch.

**Review by aarohinadkarni — APPROVED** [review]
👍

**Review by rushtong — APPROVED** [review]
Thank you for the changes 👍🏽

**Pull request merged**